"blank: nameerror: free variable 're' referenced before assignment in enclosing scope i have a strange nameerror in blank 3.3.1 (win7)...the code:..import re..# .....# parse exclude patterns..excluded_regexps = set(re.compile(regexp) for regexp in options.exclude_pattern)..# this is line 561:.excluded_regexps |= set(re.compile(regexp, re.i) for regexp in options.exclude_pattern_ci)...the error:..traceback (most recent call last):.  file ""py3createtorrent.py"", line 794, in &lt;module&gt;.    sys.exit(main(sys.argv)).  file ""py3createtorrent.py"", line 561, in main.    excluded_regexps |= set(re.compile(regexp, re.i) for regexp in options.exclude_pattern_ci).  file ""py3createtorrent.py"", line 561, in &lt;genexpr&gt;.    excluded_regexps |= set(re.compile(regexp, re.i) for regexp in options.exclude_pattern_ci).nameerror: free variable 're' referenced before assignment in enclosing scope...note that line 561, where the error occurs, is the second line in the code above. in other words: re is not a free variable. it is simply the regular expression module and it can be referenced perfectly fine in the first line...it seems to me that the reference to re.i is causing the problem, but i don't see how."
